### DISCUSSION TITLE
How does detox track the timing of the tests?

### DISCUSSION THREAD

**Discussion by AfonsoGB** [wix/Detox#3385] [Q&A]
I'm trying to figure out where the timing of each test is allocated. I would like to do something as simple as export those timings.

**Comment by julianD77** [comment]
Take a look at the Timeline Plugin that will produce a JSON artifact that is viewable in Chrome’s tracing view

https://wix.github.io/Detox/docs/api/artifacts/

**Comment by noomorph** [comment]
You can write something like this reporter and register it in your Jest config: https://github.com/jodonnell/jest-slow-test-reporter/blob/master/jest-slow-test-reporter.js

**Comment by MuckT** [comment]
Also take a look at [jest-html-reporter](https://www.npmjs.com/package/jest-html-reporter) which reports the times of test suites and individual tests. It also has the ability to set `executionTimeWarningThreshold` so that if a test suite is over the limit it appears with red text in the report. [Here's](https://github.com/valora-inc/wallet/blob/main/e2e/config.json#L9-L19) how we configured it for Detox.
